stumpy/core.py

```diff
@@ -1261,6 +1268,10 @@ def mass_absolute(Q, T, T_subseq_isfinite=None, p=2.0):
         if T_subseq_isfinite is None:
             T, T_subseq_isfinite = preprocess_non_normalized(T, m)
         distance_profile[:] = _mass_absolute(Q, T, p)
+        if query_idx is not None:  # pragma: no cover
```

Comment:
What do you think about doing:

```
if query_idx is not None:
    query_idx = int(query_idx)
    if not np.allclose(Q, T[query_idx:query_idx+m]):
        msg = "Subsequences `Q` and `T[query_idx:query_idx+m]` are different but were expected to be identical. Please verify that `query_idx` is correct."
        warnings.warn(msg)
    distance_profile[query_idx] = 0.0
```

Reply:
Yeah..I think it is a good idea! I will add it.